### PR TITLE
api: gTTA implementation (WIP)

### PR DESCRIPTION
**Pull request by thibault-martinez**
Current tests are covering failure cases.
Still need to add an extensive test on success gTTA.

# Test Plan:
Added test

### CODE REVIEW

**Review by tsvisabo — APPROVED** [review]

File: cclient/response/get_transactions_to_approve.c [hunk 1]
```diff
@@ -12,18 +12,34 @@ get_transactions_to_approve_res_t* get_transactions_to_approve_res_new() {
       (get_transactions_to_approve_res_t*)malloc(
           sizeof(get_transactions_to_approve_res_t));
   if (res) {
-    res->trunk = trit_array_new(NUM_TRITS_ADDRESS);
-    res->branch = trit_array_new(NUM_TRITS_ADDRESS);
+    res->branch = trit_array_new(HASH_LENGTH_TRIT);
```
Comment (th0br0) [inline cclient/response/get_transactions_to_approve.c]: branch & trunk are only 1 trit long, shouldn't these be fixed size `flex_trit_t` arrays?

File: ciri/api/api.c [hunk 2]
```diff
@@ -19,6 +19,12 @@
  * Private functions
  */
 
+bool invalid_subtangle_status(iota_api_t const *const api) {
```
Comment (th0br0) [inline ciri/api/api.c]: make this `static`

**Review by th0br0 — CHANGES_REQUESTED** [review]
Fix conflicts as well?

File: cclient/response/get_transactions_to_approve.c (hunk 1 above)
Comment (thibault-martinez) [reply cclient/response/get_transactions_to_approve.c]: I wanted to do that, but it has a lot of implications in the serialization; and I think @oopsmonk already has an issue regarding this.

Comment (th0br0) [reply cclient/response/get_transactions_to_approve.c]: Fair enough

**Review by tsvisabo — APPROVED** [review]

File: cclient/request/get_transactions_to_approve.c [hunk 3]
```diff
@@ -19,9 +19,22 @@ get_transactions_to_approve_req_t* get_transactions_to_approve_req_new() {
...
+  if (!req || !(*req)) {
+    return;
   }
+
+  free(*req);
+  *req = NULL;
+}
+
+void get_transactions_to_approve_req_set_depth(
+    get_transactions_to_approve_req_t* const req, uint32_t const depth) {
```
Comment (jserv) [inline cclient/request/get_transactions_to_approve.c:33]: Need indention here.

File: cclient/request/get_transactions_to_approve.c [hunk 4]
```diff
@@ -19,9 +19,22 @@ get_transactions_to_approve_req_t* get_transactions_to_approve_req_new() {
...
+  req->depth = depth;
+}
+
+void get_transactions_to_approve_req_set_reference(
+    get_transactions_to_approve_req_t* const req,
+    tryte_t const* const reference) {
+  memcpy(req->reference, reference, FLEX_TRIT_SIZE_243);
```
Comment (jserv) [inline cclient/request/get_transactions_to_approve.c:43]: Ditto. Indention.

File: cclient/request/get_transactions_to_approve.c (hunk 3 above)
Comment (thibault-martinez) [reply cclient/request/get_transactions_to_approve.c:33]: Not following our coding style.

File: cclient/request/get_transactions_to_approve.c (hunk 4 above)
Comment (thibault-martinez) [reply cclient/request/get_transactions_to_approve.c:43]: Ditto.

File: cclient/request/get_transactions_to_approve.c [hunk 5]
```diff
@@ -12,16 +12,37 @@ get_transactions_to_approve_req_t* get_transactions_to_approve_req_new() {
       (get_transactions_to_approve_req_t*)malloc(
```
Comment (th0br0) [inline cclient/request/get_transactions_to_approve.c]: This leaves depth uninitialised. Could we calloc here?

File: cclient/request/get_transactions_to_approve.c [hunk 6]
```diff
@@ -12,16 +12,37 @@ get_transactions_to_approve_req_t* get_transactions_to_approve_req_new() {
...
           sizeof(get_transactions_to_approve_req_t));
   if (req) {
-    memset(req->reference, FLEX_TRIT_NULL_VALUE, FLEX_TRIT_SIZE_243);
+    req->reference = NULL;
```
Comment (th0br0) [inline cclient/request/get_transactions_to_approve.c]: With calloc ^ not necessary

**Review by th0br0 — APPROVED** [review]

**Pull request merged**